### ISSUE TITLE
SA1600 triggers for internal interfaces when documentInterfaces = true and documentInternalElements = false

### ISSUE THREAD

**Issue by smaillet** [DotNetAnalyzers/StyleCopAnalyzers#2934]
SA1600 is mis-firing on internal interfaces when the stylecop.json documentationRules has
 documentInterfaces = true and documentInternalElements = false

example
```C#
namespace Foo
{
    internal interface Bar // SA1600 fires for this when it shouldn't
    {
        void Baz();
    }
}
```

**Comment by sharwell** [comment]
This is the intended behavior of the feature. Some projects find that documenting all internal code is overwhelming or unrealistic, and instead focuses on documenting the interfaces by which various subsystems communicate.

**Comment by smaillet** [comment]
I'm confused, how is it by design that ignore the explicit setting that I don't care about docs on internal elements. How else can I make these warning go away. "Some people" isn't all people so the tool needs to have the flexibility to do what people intend and the current options are either broken or misleading.

**Comment by sharwell** [comment]
`documentInterfaces` is evaluated before `documentInternalElements`. If you don't want warnings reported for internal interfaces, you should set it to `false`.

**Comment by zplan** [comment]
I would like, that only public interfaces have to be documented.

I currently have the settings:
    "documentationRules": {
      "documentInterfaces": false,
      "documentExposedElements": true,
      "documentInternalElements": false
    }

With this setting, I get no warning when the public interface is not documented.

@sharwell : As you have written above documentInterfaces is evaluated before documentInternalElement, I think also before documentExposedElement

How can i achieve, that  only public interfaces have to be documented?

**Comment by RobSiklos** [comment]
Yes - I agree with @zplan.

This is currently a blocker for us to switch from the legacy analyzers. We have many external interfaces, which must have documentation that adheres to all the StyleCop rules. However, for internal interfaces (of which we also have many), we don't care if any of the documentation is bad.

If we set `documentInterfaces` to `false`, then we lose all the diagnostics for public interfaces, which is not OK.

**Comment by sharwell** [comment]
@zplan @RobSiklos since the current behavior matches what was originally intended when the feature was implemented, I'm going to resolve this issue as answered. It sounds like the design does not work well for your scenarios, so please feel free to open a new issue proposing a _specific change_ to the current behavior, and make sure to account for existing users who may be happy with the current behavior (i.e. the new behavior should only be activated by a new setting so current settings do not change).

**Comment by RobSiklos** [comment]
> feel free to open a new issue proposing a _specific change_ to the current behavior, and make sure to account for existing users who may be happy with the current behavior

@sharwell Done: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3100
